### PR TITLE
Adding skeleton loading component

### PR DESCRIPTION
**Pull request by kelsiesmurphy**
- [Description](#description)
  - [Type of change](#type-of-change)
- [Testing locally](#testing-locally)
- [Checklist](#checklist)

# Description

Projects List Loading Screen

- Previously when projects/events did not load, it would simply display the text 'Loading...'
- This PR is a change to the UI to show an animated skeleton loading screen instead of the above.

Fixes # SVA-284

## Type of change

- [x] New Feature (non-breaking change which adds functionality)

# Testing locally

- To view/replicate, go to '/app/src/NativeBase/Containers/ListContainer.tsx' and change line 279 to add a '!', 
- from 
`{params?.type && listItemsToShow ? (` 
- to 
`{!params?.type && listItemsToShow ? (`
This will stop any projects loading and show the loading screen on the projects screen of the app until the '!' is removed again.

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have removed any unnecessary comments or console logging
- [ ] I have made corresponding changes to the documentation (if required)
- [x] I have addressed accessibility, if needed
- [x] I have followed best practices, e.g. NativeBase approaches and theming
- [x] I have checked the app in dark mode, if making front-end design changes
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by kelsiesmurphy** [comment]
CI Tests are failing currently, @DougieWougie I imagine this is related to your work? Anything required at the moment on my end to rectify this?

**Comment by DougieWougie** [comment]
CI is working locally so I expect for branches raised for the actions commited to main there'll be issues. I can't check until later though.

**Review by M4RC1N76 — APPROVED** [review]
Hi @kelsiesmurphy I have familiarise myself with changes introduced by you,  run the app and it works fine!! Nice work!!

**Comment by M4RC1N76** [comment]
One more thing @kelsiesmurphy do not worry about the failing test in CI I have had the same problem (well not only me).

**Comment by M4RC1N76** [comment]
***@***.*** Richardson good to know.
    On Monday, 24 April 2023 at 12:37:19 BST, Dougie Richardson ***@***.***> wrote:  
 
 


CI is working locally so I expect for branches raised for the actions commited to main there'll be issues. I can't check until later though.

File: app/src/NativeBase/Components/SkeletonLoading.tsx
Comment (joefhall) [inline app/src/NativeBase/Components/SkeletonLoading.tsx:10]: Tiny thing, but we don't need the React fragment here (`<>` and `</>`) -- [that's only needed when we don't have a single top-level component in some JSX we're rendering](https://legacy.reactjs.org/docs/fragments.html), whereas here the `VStack` does that job.

File: app/src/NativeBase/Components/SkeletonLoading.tsx
Comment (joefhall) [inline app/src/NativeBase/Components/SkeletonLoading.tsx:15]: Could we remove this `overflow` property?  It appears to work the same visually without it (unless there's something I haven't spotted)

**Comment by kelsiesmurphy** [comment]
Thanks folks! That is useful to know!

File: app/src/NativeBase/Containers/ListContainer.tsx
Comment (joefhall) [inline app/src/NativeBase/Containers/ListContainer.tsx:328]: Please could you also remove the `Text` import at the top of the file which isn't needed now

**Review by joefhall — CHANGES_REQUESTED** [review]
Nice work @kelsiesmurphy!  Just spotted a few tiny things if you could take a look

File: app/src/NativeBase/Components/SkeletonLoading.tsx
Comment (kelsiesmurphy) [reply app/src/NativeBase/Components/SkeletonLoading.tsx]: Good catch, forgot to move during refactoring to own component file.

File: app/src/NativeBase/Components/SkeletonLoading.tsx
Comment (kelsiesmurphy) [reply app/src/NativeBase/Components/SkeletonLoading.tsx]: Done!

File: app/src/NativeBase/Containers/ListContainer.tsx
Comment (kelsiesmurphy) [reply app/src/NativeBase/Containers/ListContainer.tsx:328]: Done!

File: app/src/NativeBase/Components/SkeletonLoading.tsx
Comment (kelsiesmurphy) [reply app/src/NativeBase/Components/SkeletonLoading.tsx]: Done!

**Review by joefhall — APPROVED** [review]
Lovely stuff, thanks @kelsiesmurphy!

**Comment by kelsiesmurphy** [comment]
> 

No worries @DougieWougie, should I hold off on merging until it shows as passed? Or can I merge it today?

**Pull request merged**
